x/poolmanager/types/keys.go

```diff
@@ -24,13 +25,24 @@ var (
 
 	// KeyPoolVolumePrefix defines prefix to store pool volume.
 	KeyPoolVolumePrefix = []byte{0x03}
+
+	// DenomTradePairPrefix defines prefix to store denom trade pair for taker fee.
+	DenomTradePairPrefix = []byte{0x04}
 )
 
 // ModuleRouteToBytes serializes moduleRoute to bytes.
 func FormatModuleRouteKey(poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%d", SwapModuleRouterPrefix, poolId))
 }
 
+// FormatDenomTradePairKey serializes denom trade pair to bytes.
+// Denom trade pair is automatically sorted lexicographically.
+func FormatDenomTradePairKey(denom0, denom1 string) []byte {
+	denoms := []string{denom0, denom1}
+	sort.Strings(denoms)
+	return []byte(fmt.Sprintf("%s%s%s", DenomTradePairPrefix, denoms[0], denoms[1]))
+}
```

Comment:
Would be good to have a unit test to make sure that sorting is guarded by the test to avoid accidental removal since order is instrumental to this mechanism functioning correctly

Reply:
Yeah good call, added basic key test here https://github.com/osmosis-labs/osmosis/pull/6034/commits/cbc2ff4cbc766b15b0dbcd71f622f9591b76ecc2